spec/volcanic/authenticator/v1/warden_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+RSpec.describe Volcanic::Authenticator::V1::Warden, :vcr do
+  before { Configuration.set }
+  let(:mock_request) { double 'request' }
+  let(:allow_always_strategy) { Volcanic::Authenticator::V1::Warden::AllowAlways.new(nil) }
+  let(:validate_token_always_strategy) { Volcanic::Authenticator::V1::Warden::ValidateTokenAlways.new(nil) }
+  let(:validate_token_present_strategy) { Volcanic::Authenticator::V1::Warden::ValidatePresentToken.new(nil) }
+  let(:tokens) { JSON.parse(Configuration.mock_tokens) }
+  let(:mock_token_base64) { tokens['token'] }
+  let(:mock_auth_header) { "Bearer #{mock_token_base64}" }
+  let(:invalid_auth_header) { 'invalid' }
+
+  before do
+    allow(mock_request).to receive(:headers).and_return(mock_auth_header)
+    allow(mock_request).to receive(:has_header?).with('HTTP_AUTHORIZATION').and_return(true)
+    allow(mock_request).to receive(:get_header).with('HTTP_AUTHORIZATION').and_return(mock_auth_header)
+  end
+
+  describe '#allow_always' do
+    before { allow(allow_always_strategy).to receive(:request).and_return(mock_request) }
```

Comment:
a very similar pair of lines to this are used a couple of times later...
If you add to the top something like:
```
let(:strategy) {}
subject { strategy.authenticate! }
before { allow(strategy).to receive(:request).and_return(mock_request) }
```
then these two lines could be replaced with `let(:strategy) { allow_always_strategy }`. Later on, you could then get rid of all the similar lines and replace with one-line `let(:strategy)`